### ISSUE TITLE
[BUG] formatted printing of mlx arrays broken

### ISSUE THREAD

**Issue by ngaloppo** [ml-explore/mlx#1538]
**Describe the bug**
Printing of arrays with format strings is broken, output `TypeError: unsupported format string passed to mx.array.__format__
`. 

**To Reproduce**

The following code worked in 0.19, broken in 0.19.1, presumably due to #1521

```python
import mlx.core as mx

a = mx.arange(1)
print(a) #works
print(f"{a}") #broken
```

**Expected behavior**
Expected output: 
```
array([0], dtype=int32)
array([0], dtype=int32)
```

**Comment by ngaloppo** [comment]
Thanks! For what it's worth, it is now breaking two third-party apps that use `mlx` Python module as of today (`DiffusionKit` and `ComfyUI-MLX`).

**Comment by awni** [comment]
Fix in #1539. We can do a patch release once it lands and yank 0.19.2
